lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -170,29 +172,15 @@ void StagesSubject::afterOpen()
 	// frequent, we won't implement this for now.  PPT, 22-Dec-2017.
 	std::for_each(std::begin(fStageListeners), std::end(fStageListeners),
 		[](StageListenerMap::value_type element) { TfNotice::Revoke(element.second); } );
-
-	StagesSubject::Ptr me(this);
-	for (auto stage : ProxyShapeHandler::getAllStages())
-	{
-		fStageListeners[stage] = TfNotice::Register(
-			me, &StagesSubject::stageChanged, stage);
-	}
+	fStageListeners.clear();
 
 	// Set up our stage to proxy shape UFE path (and reverse)
 	// mapping.  We do this with the following steps:
 	// - get all proxyShape nodes in the scene.
 	// - get their Dag paths.
 	// - convert the Dag paths to UFE paths.
 	// - get their stage.
-	g_StageMap.clear();
-	auto proxyShapeNames = ProxyShapeHandler::getAllNames();
-	for (const auto& psn : proxyShapeNames)
-	{
-		MDagPath dag = nameToDagPath(psn);
-		Ufe::Path ufePath = dagPathToUfe(dag);
-		auto stage = ProxyShapeHandler::dagPathToStage(psn);
-		g_StageMap.addItem(ufePath, stage);
-	}
```

Comment:
All this code was moved into the stage map. We just set it dirty now. It will be rebuilt when queried.